### ISSUE TITLE
Reading double values yields corrupted results.

### ISSUE THREAD

**Issue by chrisdew** [sidorares/nodejs-mysql-native#68]
I've built a test case:

```
mysql> select * from double_test;
+----+-------+
| id | dval  |
+----+-------+
|  1 | 1.234 | 
|  2 | 2.345 | 
+----+-------+
2 rows in set (0.00 sec)

mysql> describe double_test;
+-------+---------+------+-----+---------+----------------+
| Field | Type    | Null | Key | Default | Extra          |
+-------+---------+------+-----+---------+----------------+
| id    | int(10) | NO   | PRI | NULL    | auto_increment | 
| dval  | double  | NO   |     | 0       |                | 
+-------+---------+------+-----+---------+----------------+
2 rows in set (0.01 sec)
```

If I run the following code, I get:

```
$ node testdouble.js 
The "sys" module is now called "util". It should have a similar interface.
connected
building query select * from double_test
result { insert_id: undefined,
  affected_rows: undefined,
  rows: 
   [ { id: 1, dval: -1.7617587856133278e+42 },
     { id: 2, dval: 9.070949719158507e+135 } ] }
```

```
var my = require("mysql-native");

var db = my.createTCPClient("172.27.1.40", 3306);
db.auto_prepare = true;
db.auth("test",  "chris", "passwd");
db.addListener('connect', function() {
  console.info("connected");
  var q = buildQuery("select * from double_test");
  q(db, [], function(err, result) {
    if (err) console.log("err", err);
    if (result) console.log("result", result);
  });
});

function buildQuery(sql, debug) {
  console.log("building query", sql);
  return function(db, data, callback) {
    //console.log("executing query", sql, data);
    var query = db.execute(sql, data);
    query.on('error', function(err) {
      console.warn("error", JSON.stringify(err), sql, data);
      callback(err);
    });
    query.on('result', function(result) {
      //console.info("result", result);
      callback(null, result);
    });
  }
}
```

P.S. I forked trunk today, so I am using the latest code.  https://github.com/chrisdew/nodejs-mysql-native

**Comment by chrisdew** [comment]
I've investigated this and the problem is not with `function parseIEEE754Double(data)`, the problem is the data which is passed to it.

I amended the start of `function parseIEEE754Double(data)` to debug the issue:

```
function parseIEEE754Double(data)
{
   // debug
   var bytes = [];
   for (var i = 0; i < data.length; i++) {
      bytes[i] = data.charCodeAt(i);
   }
   console.log('bytes', JSON.stringify(bytes));
   // end debug

    var fraction = 0.0;
    fraction += data.charCodeAt(2) / ( 16 * 256 * 256 * 256 * 256 * 256 * 256);
    ...
```

and the result, when doing `select dval from double_test` was correct:

```

$ node testdouble.js 
The "sys" module is now called "util". It should have a similar interface.
connected
building query select dval from double_test
bytes [0,0,88,57,180,200,118,190,243,63]
bytes [0,0,195,245,40,92,143,194,2,64]
result { insert_id: undefined,
  affected_rows: undefined,
  rows: [ { dval: 1.234 }, { dval: 2.345 } ] }
```

When the double field is not the first field, it is corrupted:

```
$ node testdouble.js 
The "sys" module is now called "util". It should have a similar interface.
connected
building query select id, dval from double_test
bytes [0,0,1,0,0,0,88,57,180,200,118,190,243,63]
bytes [0,0,2,0,0,0,195,245,40,92,143,194,2,64]
result { insert_id: undefined,
  affected_rows: undefined,
  rows: 
   [ { id: 1, dval: -1.7617587856133278e+42 },
     { id: 2, dval: 9.070949719158507e+135 } ] }
```

Note that the size of the `data` array, which is passed to the function, _increases_ with the number of fields present.

What should be passed to `function parseIEEE754Double(data)` is simply the eight bytes of the double (and the function should be altered to not ignore the first two bytes).

**Comment by sidorares** [comment]
Thanks, Chris! I'll try to take a look tomorrow. If you already have fix, feel free to create test and pull request (they are now automatically checked by travis!)

**Comment by chrisdew** [comment]
Thanks Sidorares.  I don't have a fix, as I'm not sure what the side effects will be if I mess with these internals without a through understanding of the code.

**Comment by chrisdew** [comment]
Hi, I'm starting work on fixing this.  I'm being an idiot and haven't found the instructions on how to create the test DB and tables.  I've created 'test', with 'testuser'/'testpass', but my next error is:

```
$ node expresso-wrapper.js 
The "sys" module is now called "util". It should have a similar interface.
 starting test large integers
 starting test bit fields
 starting test select all rows row event

events.js:48
        throw arguments[1]; // Unhandled 'error' event
                       ^
Error: Table 'test.tbl' doesn't exist
```

Is there a guide to running the test suite?

**Comment by sidorares** [comment]
take a look at travis log: http://travis-ci.org/#!/sidorares/nodejs-mysql-native/jobs/1734159
this line is ftom .travis.yml file:

```
before_script:
  - "cat test/mysql_add_user.sql | mysql -uroot"
```

**Comment by sidorares** [comment]
published @ v0.4.6
